src/mlpack/tests/ann_layer_test.cpp

```diff
@@ -1165,21 +1166,37 @@ BOOST_AUTO_TEST_CASE(SimpleSigmoidCrossEntropyLayerTest)
   double error2 = module.Forward(std::move(input2), std::move(target2));
   BOOST_REQUIRE_SMALL(error2 / input2.n_elem - 1.5027283, 1e-6);
 
+  input3 = arma::mat("0 -1 -1 0 -1 0 0 -1");
+  target3 = arma::mat("0 -1 -1 0 -1 0 0 -1");
+  double error3 = module.Forward(std::move(input3), std::move(target3));
+  BOOST_REQUIRE_SMALL(error3 / input3.n_elem - 0.00320443, 1e-6);
+
   // Test the Backward function.
   module.Backward(std::move(input1), std::move(target1), std::move(output));
   for (size_t i = 0; i < output.n_elem; i++)
-    BOOST_REQUIRE_SMALL(output(i) - 0.62245929, 1e-5);
+    BOOST_REQUIRE_SMALL(output(i) - 0.37754066, 1e-5);
   BOOST_REQUIRE_EQUAL(output.n_rows, input1.n_rows);
   BOOST_REQUIRE_EQUAL(output.n_cols, input1.n_cols);
 
   expectedOutput = arma::mat(
-      "0.7310586 0.88079709 -0.04742587 0.98201376 -0.00669285");
+      "0.2689414 0.1192029 -0.04742587 0.0179862 -0.00669285");
```

Comment:
Could I ask why these numbers have changed from the original test?  I am not sure that there was anything wrong with that test.

Reply:
The previous implementation was slightly incorrect. See [here](https://github.com/mlpack/mlpack/pull/1104#discussion_r137109612)

Reply:
I think it might improve the overall readability if we define the expected results in above instead of using them inside the condition. What do you think?

Reply:
This has to be done for the other instances of tests in this sub-module I presume?

Reply:
@rcurtin The tests have reverted to original, after changing the incorrect conditional.